### ISSUE TITLE
Type Decorator Required for optional array fields

### ISSUE THREAD

**Issue by iddan-flycode** [darraghoriordan/eslint-plugin-nestjs-typed#22]
This code errors with the error: `A non-primitve property with validation should probably use a @Type decorator`

```typescript
class ExampleDto {
  @ApiPropertyOptional({
    isArray: true,
  })
  @Allow()
 exampleProperty?: string[];
}
```

while this code doesn't error:
``` typescript
class ExampleDto {
  @ApiProperty({
    isArray: true,
  })
  @Allow()
 exampleProperty!: string[];
}
```

**Comment by darraghoriordan** [comment]
Ah nice one the optionality might have changed the AST. i'll take a look at this soon. 

Out of interest, what behaviour would you expect here for primitive arrays - do you set @Type(()=> String) here or not?

**Comment by iddan-flycode** [comment]
For primitive types I rather not use the @type decorator. I think it should be required only when needed by the library to transform the DTO

**Comment by darraghoriordan** [comment]
Hey!

This is fixed in 3.9.0 - I had to slightly change the way arrays are detected with optional properties. 

I added your cases above to the test suite and it all seems ok. 

thanks for the bug report! 

dar

**Comment by iddan-flycode** [comment]
Thank you 🙏
